src/theme/components/types.ts

```diff
@@ -40,3 +42,54 @@ export type AnimationTokens = {
         | string
         | ((props: StyledComponentProps) => string);
 };
+
+export type NavbarTokens = {
+    "navbar-full-height": number | ((props: StyledComponentProps) => string);
+    "navbar-full-logo-height":
+        | number
+        | ((props: StyledComponentProps) => string);
+    "navbar-compressed-height":
+        | number
+        | ((props: StyledComponentProps) => string);
+    "navbar-compressed-logo-height":
+        | number
+        | ((props: StyledComponentProps) => string);
+    "navbar-mobile-height": number | ((props: StyledComponentProps) => string);
+    "navbar-mobile-logo-height":
+        | number
+        | ((props: StyledComponentProps) => string);
+    "navbar-colour-bg": string | ((props: StyledComponentProps) => string);
+    "navbar-colour-icon": string | ((props: StyledComponentProps) => string);
+    "navbar-link-colour-text":
+        | string
+        | ((props: StyledComponentProps) => string);
+    "navbar-link-colour-text-hover":
+        | string
+        | ((props: StyledComponentProps) => string);
+    "navbar-link-colour-text-selected-hover":
+        | string
+        | ((props: StyledComponentProps) => string);
+};
+
+export type FooterTokens = {
+    "footer-colour-bg": string | ((props: StyledComponentProps) => string);
+    "footer-colour-text": string | ((props: StyledComponentProps) => string);
+    "footer-link-colour-text":
+        | string
+        | ((props: StyledComponentProps) => string);
+    "footer-link-colour-text-hover":
+        | string
+        | ((props: StyledComponentProps) => string);
+    "footer-disclaimer-link-colour-text":
+        | string
+        | ((props: StyledComponentProps) => string);
+    "footer-disclaimer-link-colour-text-hover":
+        | string
+        | ((props: StyledComponentProps) => string);
+    "footer-disclaimer-link-colour-icon":
+        | string
+        | ((props: StyledComponentProps) => string);
+    "footer-disclaimer-link-colour-icon-hover":
+        | string
+        | ((props: StyledComponentProps) => string);
```

Comment:
icon can inherit the colour from text, so is there still a need to have separate tokens? or do we forsee a need to have differently coloured icons?